src/irc/IdentGenerator.ts

```diff
@@ -208,6 +177,31 @@ export class IdentGenerator {
         return uname;
     }
 
+    /**
+     * Get the next available suffix for a given complete username.
+     * This function will find the correct suffix for a user in as fewest
+     * DB calls as possible.
+     * @param uname The IRC username
+     * @param domain The IRC domain.
+     */
+    private async getSuffixForUsername(username: string, domain: string) {
+        let suffixLength = 1;
+        while (suffixLength < IdentGenerator.MAX_USER_NAME_SUFFIX.toString().length) {
+            const prefix = username.substr(
```

Comment:
substr is deprecated and for your parameters they behave the same.
```suggestion
            const prefix = username.substring(
```